src/code/z_message.c

```diff
@@ -365,19 +386,19 @@ void func_80151A68(PlayState* play, u16 textId) {
     }
 }
 
-void func_80151BB4(PlayState* play, u8 arg1) {
+void Message_BombersNotebookQueueEvent(PlayState* play, u8 event) {
     MessageContext* msgCtx = &play->msgCtx;
-    u8 temp = arg1;
 
     if (CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK)) {
-        if (!CHECK_WEEKEVENTREG(D_801C6B28[arg1])) {
-            msgCtx->unk120B2[msgCtx->unk120B1] = temp;
-            msgCtx->unk120B1++;
+        if (!CHECK_WEEKEVENTREG(gBombersNotebookWeekEventFlags[event])) {
+            msgCtx->bombersNotebookNewEventQueue[msgCtx->bombersNotebookNewEventQueueSize] = event;
+            msgCtx->bombersNotebookNewEventQueueSize++;
         }
-    } else if (arg1 >= 20) {
-        if (!CHECK_WEEKEVENTREG(D_801C6B28[arg1])) {
-            msgCtx->unk120B2[msgCtx->unk120B1] = temp;
-            msgCtx->unk120B1++;
+    } else if (event >= BOMBERS_NOTEBOOK_PERSON_MAX) {
+        // Non MET events are processed even if the player does not have the notebook yet
+        if (!CHECK_WEEKEVENTREG(gBombersNotebookWeekEventFlags[event])) {
+            msgCtx->bombersNotebookNewEventQueue[msgCtx->bombersNotebookNewEventQueueSize] = event;
+            msgCtx->bombersNotebookNewEventQueueSize++;
```

Comment:
```suggestion
            msgCtx->bombersNotebookNewEventQueueCount++;
```
It's more of a count instead of a size? Noting our style usually refers to size in bytes (`size_t`)